### PR TITLE
provider/aws: Support MFA delete for s3 bucket versioning

### LINKED ISSUE #7902: Feature Request for "MFA Delete" on S3 buckets
Hello! 

This is a feature request.  I'm using Terraform version v0.6.16 and would like to be able to set "MFA Delete" on S3 buckets ( as described in https://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html#MultiFactorAuthenticationDelete )
### Terraform Version

v0.6.16
### Affected Resource(s)
- aws_s3_bucket

### PR DESCRIPTION
**Pull request by stack72**
Fixes #7902

### CODE REVIEW

**Comment by catsby** [comment]
Is this still WIP?

**Comment by stack72** [comment]
Hi @catsby 

Thanks for the ping here - finally got this working now :)

P.

**Comment by Ninir** [comment]
Awesome @stack72 ! thanks :)

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 1]
```diff
@@ -147,24 +147,24 @@ func resourceAwsS3Bucket() *schema.Resource {
 			},
 
 			"versioning": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```
Comment (paddyforan) [inline builtin/providers/aws/resource_aws_s3_bucket.go]: Gonna put my ignorance on display here for a moment: would this break configs/require a migration?

Comment (stack72) [inline builtin/providers/aws/resource_aws_s3_bucket.go]: Technically yes, as we are moving from the hashcode to an int - but I figured as it was a 0.8 release and we only ever allow 1 set of versioning, that we wouldn't cause any issue here

**Comment by stack72** [comment]
Pre this PR I had a bucket that had the following state:

```
% terraform show                                                                                                                ✭
aws_s3_bucket.bucket:
  id = tf-test-bucket-paddy-test
  acceleration_status =
  acl = public-read
  arn = arn:aws:s3:::tf-test-bucket-paddy-test
  bucket = tf-test-bucket-paddy-test
  force_destroy = false
  hosted_zone_id = Z3BJ6K6RIION7M
  region = us-west-2
  request_payer = BucketOwner
  tags.% = 0
  versioning.# = 1
  versioning.2972667452.enabled = false
  website.# = 0
```

**Comment by stack72** [comment]
Post this PR, changing the set to a list has no effect:

```
% terraform plan                                                                                                                ✭
[WARN] /Users/stacko/Code/go/bin/terraform-provider-aws overrides an internal plugin for aws-provider.
  If you did not expect to see this message you will need to remove the old plugin.
  See https://www.terraform.io/docs/internals/internal-plugins.html
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but
will not be persisted to local or remote state storage.

aws_s3_bucket.bucket: Refreshing state... (ID: tf-test-bucket-paddy-test)

No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.
```

 ```
% terraform show                                                                                                                ✭
[WARN] /Users/stacko/Code/go/bin/terraform-provider-aws overrides an internal plugin for aws-provider.
  If you did not expect to see this message you will need to remove the old plugin.
  See https://www.terraform.io/docs/internals/internal-plugins.html
aws_s3_bucket.bucket:
  id = tf-test-bucket-paddy-test
  acceleration_status =
  acl = public-read
  arn = arn:aws:s3:::tf-test-bucket-paddy-test
  bucket = tf-test-bucket-paddy-test
  force_destroy = false
  hosted_zone_id = Z3BJ6K6RIION7M
  region = us-west-2
  request_payer = BucketOwner
  tags.% = 0
  versioning.# = 1
  versioning.2972667452.enabled = false
  website.# = 0
```

**Comment by stack72** [comment]
Took @paddyforan through this to clear up his question :)

**Comment by paddyforan** [comment]
Yup, questions cleared up, post-merge LGTM :)

**Pull request merged**